src/ensembl/src/content/app/browser/track-panel/track-panel-modal/modal-views/TrackPanelBookmarks.tsx

```diff
@@ -4,61 +4,150 @@ import { connect } from 'react-redux';
+import { getActiveGenomePreviouslyViewedObjects } from 'src/content/app/browser/track-panel/trackPanelSelectors';
 import { fetchExampleEnsObjects } from 'src/ens-object/ensObjectActions';
 import { getExampleEnsObjects } from 'src/ens-object/ensObjectSelectors';
 import * as urlFor from 'src/shared/helpers/urlHelper';
 import { getFormattedLocation } from 'src/shared/helpers/regionFormatter';
+import { closeTrackPanelModal } from '../../trackPanelActions';
 
 import upperFirst from 'lodash/upperFirst';
 
 import styles from '../TrackPanelModal.scss';
+import { Bookmark } from '../../trackPanelState';
 
-type TrackPanelBookmarksProps = {
+type StateProps = {
   activeGenomeId: string | null;
-  genomeInfo: GenomeInfoData;
   exampleEnsObjects: EnsObject[];
+  previouslyViewedObjects: Bookmark[];
+};
+
+type DispatchProps = {
   fetchExampleEnsObjects: (objectId: string) => void;
+  updateTrackStates: (trackStates: TrackStates) => void;
+  closeTrackPanelModal: () => void;
+};
+export type TrackPanelBookmarksProps = StateProps & DispatchProps;
+
+const getExampleObjLabel = (exampleObject: EnsObject | Bookmark) => {
+  if (exampleObject.object_type === 'gene') {
+    return exampleObject.label;
+  } else {
+    return getFormattedLocation(exampleObject.location);
+  }
+};
+
+type ExampleLinksProps = Pick<
+  TrackPanelBookmarksProps,
+  'exampleEnsObjects' | 'activeGenomeId' | 'closeTrackPanelModal'
+>;
+export const ExampleLinks = (props: ExampleLinksProps) => {
+  return (
+    <div>
+      {props.exampleEnsObjects.map((exampleObject) => {
+        const path = urlFor.browser({
+          genomeId: props.activeGenomeId,
+          focus: exampleObject.object_id
+        });
+
+        return (
+          <dd key={exampleObject.object_id}>
+            <Link to={path} onClick={props.closeTrackPanelModal}>
+              {getExampleObjLabel(exampleObject)}
+            </Link>
+            <span className={styles.previouslyViewedType}>
+              {' '}
+              {upperFirst(exampleObject.object_type)}
+            </span>
+          </dd>
+        );
+      })}
+    </div>
+  );
+};
+
+type PreviouslyViewedLinksProps = Pick<
+  TrackPanelBookmarksProps,
+  | 'previouslyViewedObjects'
+  | 'updateTrackStates'
+  | 'closeTrackPanelModal'
+  | 'activeGenomeId'
+>;
+
+export const PreviouslyViewedLinks = (props: PreviouslyViewedLinksProps) => {
+  return (
+    <div>
+      {[...props.previouslyViewedObjects]
+        .reverse()
+        .map((previouslyViewedObject, index) => {
+          const locationStr = `${previouslyViewedObject.location.chromosome}:${previouslyViewedObject.location.start}-${previouslyViewedObject.location.end}`;
+          const path = urlFor.browser({
+            genomeId: props.activeGenomeId,
+            focus: previouslyViewedObject.object_id,
+            location: locationStr
+          });
+
+          const onClickHandler = () => {
```

Comment:
please move click handler outside of the map function

Reply:
That's done